### PR TITLE
Add error message when attempting to revoke privileges from server administrator

### LINKED ISSUE #3632: No error message when trying to revoke privilege from admin or singleplayer
I know the server admin (specified in “name” config variable) is a special player in the server who has some irrevokable privileges.
However, if someone with the “privs” privilege tries to revoke such a privilege from the admin, there is no error message, it just displays the unchanged list of privileges.

So if someone tries `/revoke admin server`, an error message should be shown so the user knows at least what went wrong.

### PR DESCRIPTION
**Pull request by Splizard**

### CODE REVIEW

File: builtin/game/chatcommands.lua [hunk 1]
```diff
@@ -225,6 +225,19 @@ core.register_chatcommand("revoke", {
 			end
 		end
 		core.set_player_privs(revoke_name, privs)
+
+		--Check if we succeeded
```
Comment (est31) [inline builtin/game/chatcommands.lua]: Can you put a space after the `--`? Its mandated by the lua style guidelines: http://dev.minetest.net/Lua_code_style_guidelines

File: builtin/game/chatcommands.lua [hunk 2]
```diff
@@ -225,6 +225,19 @@ core.register_chatcommand("revoke", {
...
+		privs = core.get_player_privs(revoke_name)
+		local check_privs = revoke_privs
+		if revoke_priv_str == "all" then
+			check_privs = privs
+		end
+		for priv, _ in pairs(check_privs) do
+			if privs[priv] == true then
+				return false, "Privileges of " .. revoke_name .. " could not be revoked.. are they admin?"
```
Comment (est31) [inline builtin/game/chatcommands.lua]: before the return, can you log the attempted revocation of the privileges? Sometimes knowing who tried sth can be interesting ...

**Comment by est31** [comment]
Other than that: looks good.

**Comment by ShadowNinja** [comment]
Note that "Fix #xxx" is a bad commit message, because it requires you to look up the referenced issue to understaand the message.  Describe what the commit does without depending on an issue.  See example in the title I set.

**Comment by Splizard** [comment]
Alright I fixed the comment indentation and added a log and an updated commit message :)

File: builtin/game/chatcommands.lua [hunk 3]
```diff
@@ -225,6 +225,23 @@ core.register_chatcommand("revoke", {
...
 		end
 		core.set_player_privs(revoke_name, privs)
+
+		-- Check if we succeeded.
+		privs = core.get_player_privs(revoke_name)
+		local check_privs = revoke_privs
+		if revoke_priv_str == "all" then
+			check_privs = privs
+		end
+		for priv, _ in pairs(check_privs) do
```
Comment (kilbith) [inline builtin/game/chatcommands.lua]:
``` Lua
for priv in pairs(check_privs) do
```

File: builtin/game/chatcommands.lua [hunk 4]
```diff
@@ -225,6 +225,23 @@ core.register_chatcommand("revoke", {
...
+			if privs[priv] == true then
```
Comment (kilbith) [inline builtin/game/chatcommands.lua]:
``` Lua
if privs[priv] then
```

**Comment by Splizard** [comment]
@kilbith, I fixed those two things up and fixed up the same thing in the rest of the file. (consistency)

**Comment by kaeza** [comment]
I'm not sure this is the correct way to fix this. It looks more like a hack around deficient API.

IMHO it would be better to modify the `auth_handler.set_privileges` function to return a boolean status (plus error message in case of errors), and use that value to report if OK.

Another option would be to simply disallow privilege revocation if player whose privs are being revoked is the one named in `name` setting but the player executing the `/revoke` command isn't.

In pseudo-pseudocode:
```
local admin_name = setting_get("name")
local source_name = name
local target_name = param
if target_name==admin_name and source_name~=admin_name then
  return nil, "Nope"
end
```

**Comment by paramat** [comment]
@Splizard any thoughts on kaeza's issues?

**Comment by Splizard** [comment]
I think kaeza has a valid concern but my approach does catch any unusual circumstances where a privilege is not revoked and the player is not an admin. I don't think it is likely to happen but mods may hijack certain privileges and manage them differently.
I don't think this pull is a hack around the issue.

**Comment by rubenwardy** [comment]
#3642

**Pull request closed without merging**